test/querier.test.js

```diff
@@ -39,6 +40,110 @@ describe(`Querier: ${printPath("[test/querier.test.js]")}`, function () {
         await cleanST();
     });
 
+    // * TODO: Test that if the querier throws an error from a recipe, that recipe's ID is there
+
+    it("test that if the querier throws an error from a recipe, that recipe's ID is there", async function () {
+        await startST();
+        ST.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [Session.init()],
+        });
+        try {
+            await Session.getAllSessionHandlesForUser();
+            assert(false);
+        } catch (err) {
+            if (err.type !== ST.Error.GENERAL_ERROR || err.rId !== "session") {
+                throw err;
+            }
+        }
+    });
+
+    // * TODO: Check that once the API version is there, it doesn't need to query again
+
+    it("test that if that once API version is there, it doesn't need to query again", async function () {
+        await startST();
+        ST.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [Session.init()],
+        });
+        let q = Querier.getInstanceOrThrowError("");
+        await q.getAPIVersion();
+
+        let verifyState = await ProcessState.getInstance().waitForEvent(
+            PROCESS_STATE.CALLING_SERVICE_IN_GET_API_VERSION,
+            2000
+        );
+        assert(verifyState !== undefined);
+
+        ProcessState.getInstance().reset();
+
+        await q.getAPIVersion();
+        verifyState = await ProcessState.getInstance().waitForEvent(
+            PROCESS_STATE.CALLING_SERVICE_IN_GET_API_VERSION,
+            2000
+        );
+        assert(verifyState === undefined);
+    });
+
+    // * TODO: Check that rid is added to the header iff it's a "/recipe" || "/recipe/*" request.
+    it("test that rid is added to the header if it's a recipe request", async function () {
```

Comment:
What is the failure condition here? How have you tested that?

Reply:
@jscyo you missed this. Also, this test does not make sense to me. Please explain over a call